src/common/random_generator.cc

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2017 by Contributors
+ * \file random_generator.cc
+ * \brief cpu implements for random number generator.
+ */
+
+#include "./random_generator.h"
+
+namespace mxnet {
+namespace common {
+namespace random {
+
+template<>
+void RandGeneratorSeed<cpu, float>(Stream<cpu> *s,
+                                   RandGenerator<cpu, float> *gen,
+                                   uint32_t seed) {
+  gen->Seed(seed, 0);
+}
+
+template<>
+RandGenerator<cpu, float> *NewRandGenerator<cpu, float>() {
```

Comment:
return an value so you don't need DeleteRandGenerator

Reply:
But for `NewRandGenerator<gpu>` it is allocated on device.